### ISSUE TITLE
Doesn't compile - possibly an issue due to a recent glibc update?

### ISSUE THREAD

**Issue by WarmChocolateCake** [collectd/collectd#4034]
*   Version of collectd: 5.12.0
*   Operating system / distribution: Arch Linux
*   Kernel version (if applicable): 5.18.16

## Expected behavior

Clean compile

## Actual behavior

Compilation errors:

```
/usr/include/linux/mount.h:95:6: error: redeclaration of enum fsconfig_command
  95 | enum fsconfig_command {
     |    ^----------------
In file included from .src/utils/mount/mount.h:57,
[...]
```
## Steps to reproduce
.configure / make / make install

This appears to be something to do with glibc-2.36? (Based purely on web searches which shows a similar issue with Gentoo:
https://bugs.gentoo.org/863683

So, commenting out:

/src/utils/mount/mount.h:

```
#if HAVE_SYS_MOUNT_H
#include <sys/mount.h>
#endif

```
Resulted in a good compile.

FYI, glibc was recently upgraded in Arch linux repos (local device upgraded 3rd Aug), but I've never installed collectd before, so this may all be coincidence ;)

**Comment by ConiKost** [comment]
Yes, this is related to glibc-2.36. I am also hitting this error.

**Comment by eero-t** [comment]
It's also breaking Fedora Rawhide tests in CI.  @mrunge?

**Comment by eero-t** [comment]
This Glibc issue is best explained here: https://lore.kernel.org/all/20220802183409.GB2040@redhat.com/T/

The issue is kernel `linux/mount.h` and Glibc 2.36 `sys/mount.h` having (partially) same content, so they conflict, and  `linux/mount.h` getting also included indirectly through `linux/fs.h` (which provides some useful additional defines/macros).

Direct conflict is documented here:
* https://sourceware.org/glibc/wiki/Release/2.36#Packaging_Changes
* https://sourceware.org/glibc/wiki/Synchronizing_Headers

But how both end up being used in collectd `mount.c` happens very indirectly:
* `HAVE_XFS_XQM_H`: `mount.c` -> `xfs/xqm.h` -> `xfs/xfs.h` -> `xfs/linux.h` -> `linux/fs.h` -> `linux/mount.h`
* `HAVE_SYS_MOUNT_H`: `mount.c` -> `mount.h` -> `sys/mount.h`

I think the proposed fix of just removing inclusion for `sys/mount.h` will work only as long as `HAVE_XFS_XQM_H` happens to be defined during build, so that mount defines / enumerations are there through that other path.

=> Proper fix should exclude `sys/mount.h` only when `HAVE_XFS_XQM_H` is defined?   @mrunge?
